### PR TITLE
Add Token Factory and upgrade IBC to 7.4.0

### PR DESCRIPTION
**Pull request by ash-burnt**

### CODE REVIEW

File: integration_tests/utils.go [hunk 1]
```diff
@@ -257,6 +257,25 @@ func ModifyGenesisInflation(chainConfig ibc.ChainConfig, genbz []byte, params ..
 	return out, nil
 }
 
+func ModifyGenesisAAAllowedCodeIDs(chainConfig ibc.ChainConfig, genbz []byte, params ...string) ([]byte, error) {
+	g := make(map[string]interface{})
+	if err := json.Unmarshal(genbz, &g); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal genesis file: %w", err)
+	}
+	if err := dyno.Set(g, []int64{1}, "app_state", "abstractaccount", "params", "allowed_code_ids"); err != nil {
+		return nil, fmt.Errorf("failed to set voting period in genesis json: %w", err)
```
Comment (edjroz) [inline integration_tests/utils.go:266]:
```suggestion
		return nil, fmt.Errorf("failed to set voting allowed code ids in genesis json: %w", err)
```

File: integration_tests/utils.go [hunk 2]
```diff
@@ -257,6 +257,25 @@ func ModifyGenesisInflation(chainConfig ibc.ChainConfig, genbz []byte, params ..
...
+	}
+
+	if err := dyno.Set(g, false, "app_state", "abstractaccount", "params", "allow_all_code_ids"); err != nil {
+		return nil, fmt.Errorf("failed to set voting period in genesis json: %w", err)
```
Comment (edjroz) [inline integration_tests/utils.go:270]:
```suggestion
		return nil, fmt.Errorf("failed to set allow all code ids in genesis json: %w", err)
```

**Review by froch — APPROVED** [review]
✅

**Review by Peartes — APPROVED** [review]
LGTM

**Pull request merged**